### ISSUE TITLE
minitest raises "Unknown signal: INFO"

### ISSUE THREAD

**Issue by thbar** [oracle/truffleruby#1382]
Environment: Mac OS X 10.13, llvm@4, truffleruby 1.0.0-rc3 installed via RVM.

I get the following error when doing this:

```
$ git clone git@github.com:thbar/kiba.git truffleruby-test
$ cd truffleruby-test/
$ rvm truffleruby-1.0.0-rc3
$ gem --version
2.6.14.1
$ bundler --version
Bundler version 1.16.2
$ bundle exec rake
spawn_setup: close_others not yet implemented
Run options: --seed 44054

# Running:

/Users/thbar/.rvm/gems/truffleruby-1.0.0-rc3/gems/minitest-5.11.3/lib/minitest.rb:362:in `trap': Unknown signal: INFO (ArgumentError)
	from /Users/thbar/.rvm/gems/truffleruby-1.0.0-rc3/gems/minitest-5.11.3/lib/minitest.rb:362:in `trap'
	from /Users/thbar/.rvm/gems/truffleruby-1.0.0-rc3/gems/minitest-5.11.3/lib/minitest.rb:362:in `on_signal'
	from /Users/thbar/.rvm/gems/truffleruby-1.0.0-rc3/gems/minitest-5.11.3/lib/minitest.rb:347:in `with_info_handler'
	from /Users/thbar/.rvm/gems/truffleruby-1.0.0-rc3/gems/minitest-5.11.3/lib/minitest.rb:319:in `run'
	from /Users/thbar/.rvm/gems/truffleruby-1.0.0-rc3/gems/minitest-5.11.3/lib/minitest.rb:159:in `block in __run'
	from /Users/thbar/.rvm/gems/truffleruby-1.0.0-rc3/gems/minitest-5.11.3/lib/minitest.rb:159:in `map'
	from /Users/thbar/.rvm/gems/truffleruby-1.0.0-rc3/gems/minitest-5.11.3/lib/minitest.rb:159:in `__run'
	from /Users/thbar/.rvm/gems/truffleruby-1.0.0-rc3/gems/minitest-5.11.3/lib/minitest.rb:136:in `run'
	from /Users/thbar/.rvm/gems/truffleruby-1.0.0-rc3/gems/minitest-5.11.3/lib/minitest.rb:63:in `block in autorun'
rake aborted!
Command failed with status (1)
/Users/thbar/.rvm/gems/truffleruby-1.0.0-rc3/gems/rake-12.2.1/exe/rake:27:in `<top (required)>'
/Users/thbar/.rvm/gems/truffleruby-1.0.0-rc3/bin/truffleruby_executable_hooks:24:in `eval'
/Users/thbar/.rvm/gems/truffleruby-1.0.0-rc3/bin/truffleruby_executable_hooks:24:in `<main>'
Tasks: TOP => default => test
(See full trace by running task with --trace)
```

I have not traced what's happening further, yet.

More information on installed gems:

```
$ bundle exec gem list
spawn_setup: close_others not yet implemented

*** LOCAL GEMS ***

awesome_print (1.8.0)
bundler (1.16.2)
kiba (2.0.0)
minitest (5.11.3)
minitest-focus (1.1.2)
rake (12.2.1)
```

**Comment by eregon** [comment]
Hello, thank you for the report!

I noticed this recently as well (running MRI tests with `--native` on macOS).
A fix is being integrated and it will be in the next release, 1.0.0-rc4.

For curiosity, I do wonder why we did not discover this before, maybe we were all using Linux when running minitest?

**Comment by thbar** [comment]
You welcome! My main dev machines are all Mac OS X. Ok I'll try again after rc4 and will close here once it's confirmed!

**Comment by eregon** [comment]
@thbar As a workaround in the meantime, it is possible to just comment out the definition of the SIGINFO handler in minitest (at `/Users/thbar/.rvm/gems/truffleruby-1.0.0-rc3/gems/minitest-5.11.3/lib/minitest.rb:362`) since that's optional anyway.

**Comment by thbar** [comment]
@eregon I tried (quickly this morning) to comment out various parts of this (including just returning from `on_signal` when `name == 'INFO'`), but this affects the results of minitest, which is apparently using it for test reporting (I think - I didn't investigate at this point!).

Not a blocker for production or anything, again I'm just experimenting casually, happy to report back errors.

**Comment by eregon** [comment]
The code changed a bit, so commenting that line is indeed not going to work so well. I was looking at an older version of MiniTest when I discovered this problem.

Adding `SIGNALS.delete "INFO"` just below `SIGNALS = Signal.list` on line 350 should be a good workaround: https://github.com/seattlerb/minitest/blob/e6bc4485730403faff6966c1671cf5de72b2d233/lib/minitest.rb#L350

**Comment by eregon** [comment]
This was now fixed in SubstrateVM and will be in the next release, 1.0.0-rc4.

**Comment by eregon** [comment]
TruffleRuby 1.0.0-rc5 was released, with the fix for this bug.
@thbar Could you verify it works as expected now?

**Comment by thbar** [comment]
@eregon `bundle exec rake` works fine now, although with a large number of warnings (see below), and also `bundle install` took 5 minutes on the same repo (but this is already tracked in #1398).

Here are the logs, for the record:

```
$ bundle exec rake
/Users/thbar/.rvm/rubies/truffleruby-1.0.0-rc5/lib/ruby/gems/2.4.0/gems/bundler-1.16.3/lib/bundler/cli/exec.rb:53: warning: spawn_setup: close_others not yet implemented
/Users/thbar/.rvm/rubies/truffleruby-1.0.0-rc5/lib/truffle/rbconfig.rb:35: warning: loading in progress, circular require considered harmful - /Users/thbar/.rvm/rubies/truffleruby-1.0.0-rc5/lib/truffle/rbconfig.rb
/Users/thbar/.rvm/rubies/truffleruby-1.0.0-rc5/lib/mri/rubygems.rb:12: warning: loading in progress, circular require considered harmful - /Users/thbar/.rvm/rubies/truffleruby-1.0.0-rc5/lib/mri/rubygems.rb
/Users/thbar/.rvm/rubies/truffleruby-1.0.0-rc5/lib/mri/rubygems/version.rb:152: warning: loading in progress, circular require considered harmful - /Users/thbar/.rvm/rubies/truffleruby-1.0.0-rc5/lib/mri/rubygems/version.rb
/Users/thbar/.rvm/rubies/truffleruby-1.0.0-rc5/lib/mri/rubygems/requirement.rb:24: warning: loading in progress, circular require considered harmful - /Users/thbar/.rvm/rubies/truffleruby-1.0.0-rc5/lib/mri/rubygems/requirement.rb
/Users/thbar/.rvm/rubies/truffleruby-1.0.0-rc5/lib/mri/rubygems/platform.rb:9: warning: loading in progress, circular require considered harmful - /Users/thbar/.rvm/rubies/truffleruby-1.0.0-rc5/lib/mri/rubygems/platform.rb
/Users/thbar/.rvm/rubies/truffleruby-1.0.0-rc5/lib/mri/rubygems/specification.rb:40: warning: loading in progress, circular require considered harmful - /Users/thbar/.rvm/rubies/truffleruby-1.0.0-rc5/lib/mri/rubygems/specification.rb
/Users/thbar/.rvm/rubies/truffleruby-1.0.0-rc5/lib/mri/rubygems/core_ext/kernel_require.rb:55: warning: loading in progress, circular require considered harmful - /Users/thbar/.rvm/rubies/truffleruby-1.0.0-rc5/lib/mri/rubygems/config_file.rb
/Users/thbar/.rvm/rubies/truffleruby-1.0.0-rc5/lib/mri/rubygems/core_ext/kernel_require.rb:55: warning: loading in progress, circular require considered harmful - /Users/thbar/.rvm/rubies/truffleruby-1.0.0-rc5/lib/mri/rubygems/source.rb
/Users/thbar/.rvm/rubies/truffleruby-1.0.0-rc5/lib/mri/rubygems/core_ext/kernel_require.rb:55: warning: loading in progress, circular require considered harmful - /Users/thbar/.rvm/rubies/truffleruby-1.0.0-rc5/lib/ruby/gems/2.4.0/gems/bundler-1.16.3/lib/bundler/lockfile_parser.rb
/Users/thbar/.rvm/rubies/truffleruby-1.0.0-rc5/lib/mri/rubygems/core_ext/kernel_require.rb:55: warning: loading in progress, circular require considered harmful - /Users/thbar/.rvm/rubies/truffleruby-1.0.0-rc5/lib/ruby/gems/2.4.0/gems/bundler-1.16.3/lib/bundler/dependency.rb
/Users/thbar/.rvm/rubies/truffleruby-1.0.0-rc5/lib/mri/rubygems/core_ext/kernel_require.rb:55: warning: loading in progress, circular require considered harmful - /Users/thbar/.rvm/rubies/truffleruby-1.0.0-rc5/lib/ruby/gems/2.4.0/gems/bundler-1.16.3/lib/bundler/ruby_dsl.rb
Run options: --seed 11957

# Running:

......................................

Fabulous run in 1.250989s, 30.3760 runs/s, 49.5608 assertions/s.

38 runs, 62 assertions, 0 failures, 0 errors, 0 skips
```

**Comment by eregon** [comment]
@thbar Thank you for confirming, I can reproduce the warnings with 1.0.0-rc5.
The wrong circular require warnings are a known issue (#1402), and are already fixed on master with cdd9efc710c19b34eb9ba90dc3b084c3caf56b01 (also for kiba, I just checked).

`bundle install` only took 26s in `kiba` for me with 1.0.0-rc5 (and 28s on master).
I guess this might be due to having more installed gems on your side (maybe RVM confuses gems of rc3 and rc5? What does `gem list` show?), or maybe less free memory available leading to swapping.
